v2/oauth/multiple-frontends-with-separate-backends.mdx

```diff
@@ -0,0 +1,427 @@
+
+
+<BackendSDKTabs>
+<TabItem value="nodejs">
+
+```typescript
+import supertokens from "supertokens-node";
+import OAuth2Provider from"supertokens-node/recipe/oauth2provider";
+
+supertokens.init({
+    framework: "express",
+    supertokens: {
+        connectionURI: "<YOUR_CONNECTION_URI>",
+        apiKey: "<YOUR_API_KEY>",
+    },
+    appInfo: {
+        appName: "",
+        apiDomain: "<YOUR_API_DOMAIN>",
+        websiteDomain: "<YOUR_WEBSITE_DOMAIN>",
+        apiBasePath: "/auth",
+        websiteBasePath: "/auth"
+    },
+    recipeList: [
+        // This is where you initialize the OAuth2 recipe
+        OAuth2Provider.init(),
+    ]
+});
+```
+
+</TabItem>
+
+<TabItem value="go">
+
+:::caution
+
+At the moment we do not have support creating OAuth2 providers in the Go SDK.
+
+:::
+
+</TabItem>
+
+<TabItem value="python">
+
+:::caution
+
+At the moment we do not have support creating OAuth2 providers in the Python SDK.
+
+:::
+
+</TabItem>
+
+</BackendSDKTabs>
+
+## 3. Set Up Session Handling in Each Application
+
+You will have to setup custom session handling logic in each of your `backend applications`.
+
+<BackendSDKTabs>
+
+<TabItem value="nodejs">
+
+You can use a generic OIDC or OAuth2 library. 
+For `NodeJS` we recommend using either [passport-oauth2](https://www.passportjs.org/packages/passport-oauth2/) or [passport-openidconnect](https://www.passportjs.org/packages/passport-openidconnect/).
+
+#### Example using `passport-oauth2`:
+
+```typescript
+import express, { Request, Response } from 'express';
+import passport from 'passport';
+import { Strategy as OAuth2Strategy } from 'passport-oauth2';
+import session from 'express-session';
+
+const AUTHORIZATION_SERVER_URL = 'https://auth.example.com';
+// The client ID returned from the first step
+const CLIENT_ID = 'your-client-id';
+// The client secret returned from the first step
+const CLIENT_SECRET = 'your-client-secret';
+// On of the callback URLs specified in the create client payload 
+const CALLBACK_URL = 'https://your-app.com/auth/callback';
+
+const app = express();
+
+app.use(session({
+  secret: 'your-session-secret',
+  resave: false,
+  saveUninitialized: false
+}));
+
+app.use(passport.initialize());
+app.use(passport.session());
+
+passport.serializeUser((user, done) => {
+  done(null, user);
+});
+
+passport.deserializeUser((obj, done) => {
+  done(null, obj);
+});
+
+passport.use(new OAuth2Strategy({
+  authorizationURL: `${AUTHORIZATION_SERVER_URL}/authorize`,
+  tokenURL: `${AUTHORIZATION_SERVER_URL}/token`,
+  clientID: CLIENT_ID,
+  clientSecret: CLIENT_SECRET,
+  callbackURL: CALLBACK_URL
+}, (accessToken, refreshToken, profile, done) => {
+  // Verify if the user exists
+  // And then store the user's profile during this step
+  done(null, profile);
+}));
+
+app.get('/auth', passport.authenticate('oauth2'));
+app.get('/auth/callback', passport.authenticate('oauth2', {
+  failureRedirect: '/login'
```

Comment:
why are these required?

Reply:
Maybe instead of just adding code here like this, we should properly explain the integration. Cause this is a VERY important part, and this code here doesn't really explain or help anything. This will need a detailed guide for each language or framework. 

We can even just find generic oauth guides for each of the frameworks and link to those instead, and then add comments here saying what the values of various things like AUTHORIZATION URL and stuff should be that are referenced from that tutorial. Not sure. But this needs to be way more well explained.

Reply:
I think these are from a relevant [passport strategy guide](https://www.passportjs.org/packages/passport-oauth2/), but I agree it could bear with some expanded explanation. Although this part is fairly framework specific.